### PR TITLE
Collapse incident-handling as much as possible into single, linear doc

### PR DESCRIPTION
**Pull request by joeyAghion**
Our documentation of incident-handling procedures and tools have proliferated and it's been difficult to keep them up-to-date with evolving tools and conventions. This [work-in-progress] update attempts to consolidate them into a single doc. It doesn't yet remove the pre-existing docs.

For instructions that are _occasionally_ relevant, but shouldn't get in the way of an on-call engineer who may be under pressure to respond to an incident, I employ the `<details><summary>...` syntax to keep that content collapsed by default.

(Some of the unrelated edits and questionable whitespace choices are required by the pre-commit hook, not me.)

### CODE REVIEW

File: playbooks/incident-handling.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+This objective only relates to the time between an incident being reported and the on-call engineers' initial
+acknowledgement (and tracking in Opsgenie). We don't guarantee recovery times, though obviously those are of great
+interest to us.
+
+#### Monitoring automated alerts
+
+The [#alerts](https://artsy.slack.com/archives/alerts) channel receives automated notifications from various
+sources (Datadog, Pingdom, Sentry). It aims to be quiet and restricted to actionable events, but your judgment is
+required to assess their severity.
```
Comment (MounirDhahri) [inline playbooks/incident-handling.md:264]: **thought:** are you intentionally not linking to our old severity docs here?

**Review by MounirDhahri — APPROVED** [review]
Thanks for pulling everything togethe in one document. This makes things much easier

File: playbooks/incident-handling.md (hunk 1 above)
Comment (joeyAghion) [reply playbooks/incident-handling.md:264]: Sort of intentional. There are 2 types of severity discussed in these docs:
* Here, whether a problem is severe enough to be considered an incident needing on-call support at all. I consider this very important and that's why there are examples below and the repeated, bold rule of thumb about "significant proportion of users, critical business functions, or sensitive data."
* [Those severity docs](https://github.com/artsy/README/blob/43c400d81ff9fee7276c3dd934de26b985da362f/playbooks/support/responding.md#severity-of-incidents), which suggest how to assign priority to something which _is_ deemed an incident. I didn't find those helpful, since they're long and the priority designation ultimately doesn't affect anything.

BUT I wasn't sure about this... if you think some rules of thumb would help, I can bring them back in a collapsed and hopefully more concise section.

Comment (MounirDhahri) [reply playbooks/incident-handling.md:264]: I also found the severity docs that you linked a bit confusing, I only wanted to mention it in case you missed them. Thanks for explaining

File: playbooks/incident-handling.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+1. Click "Add" and verify your override:
+
+      <img width=500 src="images/opsgenie-override-2.png" />
+   </details>
+
+<details>
+  <summary>Special holiday shifts</summary>
+
+Holiday on-call shifts will take place from December 21st to Janurary 6th, these holiday rotations are labeled
```
Comment (sweir27) [inline playbooks/incident-handling.md:124]: #minor but since I imagine the actual dates here will change based on when Christmas falls during the week, we could make this more generic to avoid confusion.

File: playbooks/incident-handling.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+  <img width="500" src="https://user-images.githubusercontent.com/12748344/99718836-82df1700-2a79-11eb-8cba-b642f93a434b.png">
+
+3. Refer to _Trading or Overriding_ (above) to assign yourself a shift.
+   </details>
+
+<details>
+  <summary>Creating a new on-call schedule</summary>
+
+We use a staggered schedule to retain context and limit disruption as engineers shift in/out of rotation. "A"
+rotations start/end on a Monday at 11 am. "B" rotations start/end on a Wednesday at 11 am. Each engineer will be
```
Comment (sweir27) [inline playbooks/incident-handling.md:143]: Let's specify timezone here (11am ET).

File: playbooks/incident-handling.md (hunk 1 above)
Comment (sweir27) [reply playbooks/incident-handling.md:264]: If it helps, I think in the past we were exploring the possibility of creating incidents more freely vs. only when the issue is high-severity. If that were the case, the 5 levels are important.

File: playbooks/incident-handling.md [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+recording relevant information (insights, links...) on the incident timeline. If both engineers aren't available,
+the incident commander can perform both roles or ask for another volunteer.
+
+#### Diagnose
+
+Discuss the incident's resolution in the dedicated Slack channel. Tips:
+
+- Scan [recent releases](https://github.com/pulls?q=is%3Amerged+is%3Apr+archived%3Afalse+org%3Aartsy+deploy) that
+  may be responsible.
+  [Roll back](https://github.com/artsy/hokusai/blob/master/docs/Command_Reference.md#how-to-do-a-rollback) the
```
Comment (sweir27) [inline playbooks/incident-handling.md:352]: We could add a line here to help on-call engineers decide whether to roll back or revert the change as a first action, since this has been a sticking point in the past.

File: playbooks/incident-handling.md [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+
+#### Escalate
+
+When investigating an incident, you should involve the relevent team and stakeholders involved as soon as possible.
+**You are not expected to know the ins and outs of every system, so don't hesitate to involve the wider team.**
+
+<details>
+  <summary>To escalate an issue:</summary>
+
+- Identify the engineer responsible for the associated pull request or release. If not obvious, the
```
Comment (sweir27) [inline playbooks/incident-handling.md:380]: We should add a line here about the first line of escalation being the Team x Scope list-- i.e. if there's an issue with payments, we want the person to contact the Transact team first, not necessarily the Exchange point-people.

File: playbooks/incident-handling.md [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+- Add the PM as a stakeholder to the issue. This will notify the stakeholder via email and allow them to receive
+  updates on the ongoing progress of the incident.
+
+  <img width=800 src="images/opsgenie-responding-0.png" />
+
+- Alert the relevent team via slack in the #inc-\* slack channel you created for the incident and share any
+  applicable work-arounds or talking points in the thread to unblock teammates or partners.
+  </details>
+
+#### Communicate
```
Comment (sweir27) [inline playbooks/incident-handling.md:397]: It may be clearer if this section is organized around the three places the comms officer needs to be aware of/updating vs. just listing them in separate paragraphs:
1. The OpsGenie timeline
2. Slack: The original #incidents thread _and_ if discussion & resolution is happening in zoom, consider updating the #inc- channel as well with initial findings/explorations so people can follow async.
3. StatusPage

**Review by sweir27 — APPROVED** [review]
Woo looks great! Just a few comments.

I am about to go on-call and after reading this I feel 👍 😄.

File: playbooks/incident-handling.md [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+  relating to specific system deployments.
+- See also other metrics and dashboards (AWS, MongoDB, Elasticsearch/kibana, Sendgrid...).
+
+If a mitigation (such as rolling back a release) requires some follow-up, make sure to note it on the incident and
+surface to the appropriate team. If a mitigation isn't immediately apparent, **escalate to an appropriate team or
+engineer**.
+
+#### Escalate
+
+When investigating an incident, you should involve the relevent team and stakeholders involved as soon as possible.
```
Comment (ovasdi) [inline playbooks/incident-handling.md:374]:
```suggestion
When investigating an incident, you should involve the relevant team and stakeholders involved as soon as possible.
```
typo

File: playbooks/incident-handling.md [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+  [Subject Matter Expert](https://support.atlassian.com/opsgenie/docs/define-incident-response-roles/). (This can
+  also be done directly from the OpsGenie-created Slack channel.) Adding the responder will notify them and keep
+  them alerted to ongoing developments.
+
+- Add the PM as a stakeholder to the issue. This will notify the stakeholder via email and allow them to receive
+  updates on the ongoing progress of the incident.
+
+  <img width=800 src="images/opsgenie-responding-0.png" />
+
+- Alert the relevent team via slack in the #inc-\* slack channel you created for the incident and share any
```
Comment (ovasdi) [inline playbooks/incident-handling.md:393]:
```suggestion
- Alert the relevant team via slack in the #inc-\* slack channel you created for the incident and share any
```

**Review by ovasdi — COMMENTED** [review]
Left a few nitpicks/edits.

File: playbooks/incident-handling.md (hunk 1 above)
Comment (joeyAghion) [reply playbooks/incident-handling.md:264]: We should still flesh this out. (E.g., no reason we can't use "low" or "informational" priorities in a way that avoids an on-call response.) I put a to-do in the doc.

File: playbooks/incident-handling.md (hunk 5 above)
Comment (joeyAghion) [reply playbooks/incident-handling.md:382]: Let me know what you think of the updates.

File: playbooks/incident-handling.md [hunk 9]
```diff
@@ -0,0 +1,458 @@
+---
+title: Incident Handling
+description: How engineers share on-call duties and provide urgent support
+---
+
+# Incident Handling
```
[the remark this answers is not in the document]
Comment (joeyAghion) [reply playbooks/incident-handling.md:6]: I'm somewhat intentionally using the broadest possible meaning of "incident handling" here. E.g., we _should_ expand on how we handle non-urgent incidents (that don't necessarily trigger an on-call response), and when we do I'd hope that would be covered in this doc as well.

I'd be happy to update the title to "Incident Handling Procedures" if you think that's a little more meaningful.

File: playbooks/incident-handling.md [hunk 10]
```diff
@@ -0,0 +1,458 @@
...
+
+   - If outstanding questions or opportunities need further consideration, this is a good place to note them for
+     discussion in the incident review meeting.
+   - Aim to fill out the postmortem template within ~2 days.
+   - Once the basics are reflected in the postmortem, invite input on the draft from other on-call engineers or
+     subject matter experts.
+
+5. Add a new row to the
+   [Incident Review Schedule](https://www.notion.so/artsy/852e6a28fcda46ada5596f6190c9e006?v=210da6a9fe314893babbd81220b0523e)🔒
+   with a link to the postmortem and a brief description of the incident.
```
[the remark this answers is not in the document]
Comment (joeyAghion) [reply playbooks/incident-handling.md:449]: I don't see links to incidents from that list though, only to the postmortem (which seemed fine, since the postmortem is linked to the incident).

[the remark this answers is not in the document]
Comment (ovasdi) [reply playbooks/incident-handling.md:449]: The incident number is actually a hyperlink. If you click on the number twice it will take you to the incident page.

**Review by ovasdi — APPROVED** [review]

**Comment by joeyAghion** [comment]
Thanks for all the feedback. I'll merge, noting these areas of possible improvement:
* clarify how to identify additional responders, especially once projects' point people are updated and spread around geographies
* use incident priorities more strategically, perhaps by encouraging tracking of "low"/"informational" incidents that might not trigger on-call notifications
